File: Python/NarcissisticNumber.py
def narcissistic_number(number: int) -> bool:
    digits = []
    original_number = number
    while number > 0:
        digits.append(number % 10)
        number //= 10
    num_digits = len(digits)
    sum_of_powers = sum(digit ** num_digits for digit in digits)
    return original_number == sum_of_powers

print(153,":",narcissistic_number(153))
print(154,":",narcissistic_number(154))
print(1634,":",narcissistic_number(1634))
print(1635,":",narcissistic_number(1635))

File: CPP/NarcissisticNumber.cpp
#include <iostream>
#include <cmath>
#include <vector>
using namespace std;

//O(n)
bool narcissisticNumber(int number) {
    vector<int> digits;
    int originalNumber = number;
    while (number > 0) {
        digits.push_back(number % 10);
        number /= 10;
    }
    int numDigits = digits.size();
    int sumOfPowers = 0;
    for (int i=0; i < digits.size(); i++) {
        sumOfPowers += pow(digits[i], numDigits);
    }
    return originalNumber == sumOfPowers;
}

int main() {
    cout << 153 << " : " << narcissisticNumber(153) << endl;
    cout << 154 << " : " << narcissisticNumber(154) << endl;
    cout << 1634 << " : " << narcissisticNumber(1634) << endl;
    cout << 1635 << " : " << narcissisticNumber(1635) << endl;
}